ddl/constraint.go

```diff
@@ -355,13 +355,13 @@ func (w *worker) verifyRemainRecordsForCheckConstraint(dbInfo *model.DBInfo, tab
 	// It's no need to construct expression node out and pull the chunk rows through it. Here we
 	// can let the check expression restored string as the filter in where clause directly.
 	// Prepare internal SQL to fetch data from physical table under this filter.
-	sql := fmt.Sprintf("select count(1) from `%s`.`%s` where not %s limit 1", dbInfo.Name.L, tableInfo.Name.L, constr.ExprString)
+	sql := fmt.Sprintf("select 1 from `%s`.`%s` where not %s limit 1", dbInfo.Name.L, tableInfo.Name.L, constr.ExprString)
```

Comment:
Ummm, could you please tell me in which case can avoid scan whole table. 
And in my opinion, `select 1` can surly avoid aggregation like this: 
```
mysql> explain select count(1) from test.t where not a > 0 limit 1;
+--------------------------------+----------+-----------+---------------+---------------------------------+
| id                             | estRows  | task      | access object | operator info                   |
+--------------------------------+----------+-----------+---------------+---------------------------------+
| Limit_11                       | 1.00     | root      |               | offset:0, count:1               |
| └─HashAgg_19                   | 1.00     | root      |               | funcs:count(Column#4)->Column#3 |
|   └─TableReader_20             | 1.00     | root      |               | data:HashAgg_12                 |
|     └─HashAgg_12               | 1.00     | cop[tikv] |               | funcs:count(1)->Column#4        |
|       └─Selection_18           | 3323.33  | cop[tikv] |               | le(test.t.a, 0)                 |
|         └─TableFullScan_17     | 10000.00 | cop[tikv] | table:t       | keep order:false, stats:pseudo  |
+--------------------------------+----------+-----------+---------------+---------------------------------+
6 rows in set (0.00 sec)

mysql> explain select 1 from test.t where not a > 0 limit 1;
+--------------------------------+---------+-----------+---------------+--------------------------------+
| id                             | estRows | task      | access object | operator info                  |
+--------------------------------+---------+-----------+---------------+--------------------------------+
| Projection_7                   | 1.00    | root      |               | 1->Column#3                    |
| └─Limit_8                      | 1.00    | root      |               | offset:0, count:1              |
|   └─TableReader_13             | 1.00    | root      |               | data:Limit_12                  |
|     └─Limit_12                 | 1.00    | cop[tikv] |               | offset:0, count:1              |
|       └─Selection_11           | 1.00    | cop[tikv] |               | le(test.t.a, 0)                |
|         └─TableFullScan_10     | 3.01    | cop[tikv] | table:t       | keep order:false, stats:pseudo |
+--------------------------------+---------+-----------+---------------+--------------------------------+
6 rows in set (0.01 sec)
```
this also helps

Reply:
```
select count(1) from test.t where not a > 0 limit 1;
will count the whole table before the sql return.

while, select 1 from test.t where not a > 0 limit 1;
will return when meet one row satisfy the where contition.

```
so, it can report error when the sql find a row not satisfy check constraint.

Reply:
Hi, have I explained this clearly? ^_^